### DISCUSSION TITLE
How much of PB is actually using heuristics for blocking?

### DISCUSSION THREAD

**Discussion by OTonGitHub** [EFForg/privacybadger#3033] [Q&A]
I just wanted to ask how PB actually works, compared to something like uBlock origin today, I read the article on Google Security Team reporting on the vulnerability. So Instead of going through the code, since I am just a user, i thought I'd ask here to get a quick answer if that's possible.
I also read that, by default, the heuristics is off and just relies on similar mechanism like uBlock?

I like the idea of PB because, I don't necessarily want to block ads, but I want to make advertisers respect privacy by blocking ads that uses trackers. And I like the idea of using heuristics too, not relying on a list.

Also what's the difference between local learning, remote learning in terms of privacy/security etc, how does that compare to other popular anti trackers.

**Comment by OTonGitHub** [comment]
Actually.. I think this article I just found https://www.eff.org/deeplinks/2023/10/privacy-badger-learns-block-ever-more-trackers
answered most of my questions.

I am still torn between just privacy badger and uBlock. I mean, at the end of the day, both are just using lists? It's a matter of how the list was created for both sources and when you last updated the browser extension?

**Reply by ghostwords** [reply]
You could try using both extensions together and see how it goes for you. [Our FAQ](https://privacybadger.org/#Is-Privacy-Badger-compatible-with-other-extensions%2c-including-other-adblockers) goes into the differences a bit, but the gist is that uBO and PB are different tools that can work together towards a common purpose. Either one is enough by itself too, it's all good.

**Comment by ghostwords** [comment]
Thanks for finding [our latest explanation](https://www.eff.org/deeplinks/2023/10/privacy-badger-learns-block-ever-more-trackers) of PB's evolution!

We also have the following Privacy Badger FAQ entry:

[privacybadger.org > FAQs > Is Privacy Badger compatible with other extensions, including adblockers?](https://privacybadger.org/#Is-Privacy-Badger-compatible-with-other-extensions%2c-including-other-adblockers)

Let me know if you want me to go into any of the points in the FAQ in more detail.

**Reply by OTonGitHub** [reply]
Understood, I understand that the swarm is used to train and update the tracker list PB uses, and it's obviously done remotely. My question is, how frequently and up-to-date will my extension be, I am sure the list grows everyday, and I also know adding every new entry in real time probably doesn't have much of an effect on how much I am being tracked. But I am curious as to how it works, maybe I am understanding it wrong? I really like the concept, but how often does my badger get in touch with the swarm HQ is basically my question.

**Reply by ghostwords** [reply]
The list typically gets updated on Privacy Badger updates, although the last few PB updates kept the same list. We're going to do a big list refresh in the next few weeks.

Privacy Badger is more than a list though! Here are a few highlights:

- https://www.eff.org/deeplinks/2024/01/privacy-badger-puts-you-control-widgets
- https://www.eff.org/deeplinks/2023/09/new-privacy-badger-prevents-google-mangling-more-your-links-and-invading-your
- https://www.eff.org/gpc-privacy-badger
